File: BackendApi/Models/Card/QuestionAnswer.cs
namespace BackendApi.Models.Card;

public class QuestionAnswer
{
    // Id вопрос-ответа
    public int Id { get; set; }
    // Вопрос
    public string Question { get; set; }
    // Список ответов и фотографий в BASE64
    public string Answers { get; set; }
    // Пустой, если это теоретический вопрос
    public string GoodAnswersId { get; set; }
}

File: lib/database/questionanswer.dart
import 'package:hive_flutter/hive_flutter.dart';

part 'questionanswer.g.dart';

@HiveType(typeId: 1)
class QuestionAnswer {
  QuestionAnswer({
    required this.id,
    required this.question,
    required this.answers,
    required this.goodAnswersId,
  });

  @HiveField(0)
  late int id;

  @HiveField(1)
  late String question;

  @HiveField(2)
  late List<String> answers;

  @HiveField(3, defaultValue: -1)
  late int goodAnswersId;
}
